### PR TITLE
Blaze: Display overlay only once

### LINKED ISSUE #20251: Blaze Overlay: Show overlay once
## Description 

Ref: p1677787613872359-slack-C04LJFS1G5P

Currently we show the Blaze overlay every time a Blaze entry point is tapped. Update the overlay logic so that it's shown only once.

### PR DESCRIPTION
**Pull request by hassaanelgarem**
Closes #20251

## Description
This PR changes the blaze overlay logic to be only shown once per entry point. After that, the blaze web flow is presented directly.

## Testing Instructions

1. Switch to a Blaze-eligible site
2. Switch to the "Home" tab
3. Tap on the Blaze dashboard card
4. ✅ A Blaze overlay should be displayed
5. Dismiss the overlay
6. Tap on the Blaze dashboard card again
7. ✅ A Blaze webview opens
8. Repeat the above steps for the menu item, posts list, and pages list entry points

## Regression Notes
1. Potential unintended areas of impact
Features removal overlays frequency

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Testing manually and depended on `OverlayFrequencyTrackerTests`

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by mokagio** [comment]
I've bumped this to 22.0 because I'm about to start the 21.9 code freeze. 

Feel free to revert and update to target the release branch if this needs to be part of that release.

BTW, looks like the tests need to be updated, too:

image

File: WordPress/Classes/Utility/OverlayFrequencyTracker.swift [hunk 1]
```diff
@@ -93,7 +93,19 @@ class JetpackOverlayFrequencyTracker {
...
+extension OverlayFrequencyTracker {
+
+    enum FrequencyType {
+        case showOnce
+        case alwaysShow
+        case respectFrequencyConfig
+    }
+
+    enum OverlayType: String {
+        case featuresRemoval = "" // Empty string to make sure the generated keys are backwards compatible
```
Comment (momo-ozawa) [inline WordPress/Classes/Utility/OverlayFrequencyTracker.swift:105]: Gotcha, thanks for clarifying

File: WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController.m [hunk 2]
```diff
@@ -1670,10 +1670,10 @@ - (void)showBlaze
...
     [BlazeEventsTracker trackEntryPointTappedFor:BlazeSourceMenuItem];
     
-    [BlazeOverlayCoordinator presentBlazeOverlayInViewController:self
-                                                          source:BlazeSourceMenuItem
-                                                            blog:self.blog
-                                                            post:nil];
+    [BlazeFlowCoordinator presentBlazeInViewController:self
+                                                        source:BlazeSourceMenuItem
+                                                        blog:self.blog
+                                                        post:nil];
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController.m]: [nit] alignment

**Review by momo-ozawa — APPROVED** [review]
Works as described!

**Comment by hassaanelgarem** [comment]
Thanks for the heads up @mokagio!
It would be great if this is included in 21.9. I updated the milestone and the base branch 👍

File: WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController.m (hunk 2 above)
Comment (hassaanelgarem) [reply WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController.m]: Fixed in 885a79e62e062dd8d7699022bbea5d65aefd004d 👍

**Comment by AliSoftware** [comment]
This PR has now been included in the `21.9.0.1` beta.

**Pull request merged**